File: Tercera Entrega/sdi3-37.Cli-REST/src/com/sdi/presentation/user/action/CrearTareaAction.java
package com.sdi.presentation.user.action;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.client.Entity;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import com.sdi.dto.Task;
import com.sdi.presentation.user.Authenticator;
import com.sdi.presentation.user.Sesion;

import alb.util.console.Console;
import alb.util.menu.Action;

public class CrearTareaAction implements Action {

	public CrearTareaAction() {}
	
	@Override
	public void execute() throws Exception {
		Console.println("------CREAR TAREA------");
		SimpleDateFormat formateador = new SimpleDateFormat("yyyy-MM-dd");
		
		//Pedir datos
		String nombre = Console.readString("Nombre de la tarea");
		String comentario = Console.readString("Comentario de la tarea");
		comentario = comentario.replace(":", "_");
		String fecha = Console.readString("Fecha planeada(yyyy-mm-dd)");
		
		//Creamos tarea con los datos
		Task t = new Task();
		t.setUserId(Sesion.getInstance().getUser().getId());
		t.setTitle(nombre);
		t.setComments(comentario);
		t.setCreated(formateador.parse(formateador.format(new Date())));
		t.setFinished(null);
		try {
			t.setPlanned(formateador.parse(fecha));
			
			//Comprobamos si la fecha planeada esta bien
			if(t.getPlanned().compareTo(t.getCreated()) >= 0) {
				Response response = ClientBuilder.newClient()
						.register(new Authenticator(Sesion.getInstance().getUser().getLogin(), Sesion.getInstance().getUser().getPassword()))
						.target(Sesion.getInstance().getRestServiceUrl())
						.path("crearTarea")	
						.request()
						.post(Entity.entity(t, MediaType.APPLICATION_JSON));
				
				if(response.getStatus() == 204) {
					Console.println("\tSe ha creada la tarea correctamente");
				} 
				else {
					Console.println("\tError en la petición de crear tarea");
				}
			}
			else {
				Console.println("\tFecha planeada menor que fecha actual, no se crea la tarea");
			}				
		} catch(ParseException e){
			Console.println("\tFormato de fecha introducido incorrecto");
		}
		Console.println("-----------------------");
	}
}

File: Tercera Entrega/sdi3-37.Cli-REST-C_Sharp/sdi3-37.Cli-REST-C_Sharp/action/CrearTareaAction.cs
using sdi3_37.Cli_REST_C_Sharp.dto;
using sdi3_37.Cli_REST_C_Sharp.util;
using System;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_37.Cli_REST_C_Sharp.action
{
    class CrearTareaAction : Action
    {

        public CrearTareaAction ()
        {

        }

        public void execute()
        {
            Console.WriteLine("------CREAR TAREA------");

            //Pedir datos 
            Console.Write("Nombre de la tarea: ");
            var nombre = Console.ReadLine();

            Console.Write("Comentario de la tarea: ");
            var comentario = Console.ReadLine();
            comentario = comentario.Replace(":", "_");

            Console.Write("Fecha planeada(yyyy-mm-dd): ");
            var fecha = Console.ReadLine();           

            //Creamos la tarea con los datos
            Tarea t = new Tarea();
            t.userId = Sesion.Instance.User.id;
            t.title = nombre;
            t.comments = comentario;
            t.created = DateTime.Today;
            try
            {
                t.planned = Convert.ToDateTime(fecha);

                if(DateTime.Compare(t.planned,t.created) >= 0)
                {
                    insertarTarea(t).Wait();                   
                }
                else
                {
                    Console.WriteLine("\tFecha planeada menor que fecha actual, no se crea la tarea");
                }
            }
            catch (Exception e)
            {
                Console.WriteLine("\tFormato de fecha introducido incorrecto");
            }

            Console.WriteLine("-----------------------");
        }

        static async Task insertarTarea (Tarea tarea)
        {
            using (var cliente = new HttpClient())
            {
                cliente.BaseAddress = new Uri(Sesion.Instance.URL);
                cliente.DefaultRequestHeaders.Accept.Clear();
                cliente.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

                var json = Util.parsearJson(tarea);
                var stringContent = new StringContent(json, Encoding.UTF8, "application/json");
                HttpResponseMessage response = await cliente.PostAsync("crearTarea", stringContent);
                if (response.IsSuccessStatusCode)
                {
                    Console.WriteLine("\tSe ha creada la tarea correctamente");
                }
                else
                {
                    Console.WriteLine("\tHubo algún problema creando la tarea");
                }
            }
        }
    }
}
